### PR TITLE
boards: purge gnrc_netdev_default from Makefile.dep

### PR DESCRIPTION
**Pull request by benpicco**
### Contribution description

`gnrc_netdev_default` will pull in `netdev_default`, so no need to check for both in `Makefile.dep`

### Testing procedure

The default network interface should still be loaded.

### CODE REVIEW

File: boards/common/nrf52/nrf52832/Makefile.dep [hunk 1]
```diff
@@ -1,3 +1,4 @@
+# nimble_netif has a hard dependency on gnrc_netdev
```
Comment (fjmolinas) [inline boards/common/nrf52/nrf52832/Makefile.dep:1]: Is this related?

Comment (benpicco) [reply boards/common/nrf52/nrf52832/Makefile.dep:1]: Insofar that I first tried to replace this with ‘gnrc_netdev_default’ too, but Murdock found problems with the lwip tests.

Comment (fjmolinas) [reply boards/common/nrf52/nrf52832/Makefile.dep:1]: Yes but I feel that this is unrelated here, checking for `netdev_default` and `gnrc_netdev_default` is redundant, but here it is not done to include a `netdev` interface, but an extra dependency kind of what is done in Makefile.dep:

```
ifneq (,$(filter gnrc_netdev_default,$(USEMODULE)))
  USEMODULE += netdev_default
  USEMODULE += gnrc_netif
endif
```

IMO you can remove this comment, and re-phrase the commit so its not a `purge` just a remove duplicates?

Comment (benpicco) [reply boards/common/nrf52/nrf52832/Makefile.dep]: Alright then!
